client/src/main/java/io/pravega/client/control/impl/ControllerResolverFactory.java

```diff
@@ -201,7 +201,7 @@ public void start(Listener listener) {
         public void shutdown() {
             shutdown = true;
             if (scheduledFuture != null) {
-                scheduledFuture.cancel(false);
+                scheduledFuture.cancel(true);
```

Comment:
FYI, this has no effect. It is not implemented in the JRE.

Reply:
This is a ScheduledFuture on which cancel() is invoked.
` public ScheduledFuture<?> schedule(Runnable command, long delay, TimeUnit unit);`
As I understand your comment applies only on a CompletableFuture. let me know...

Reply:
Ah I missed that one. Nevermind then :)